diff --git a/configure b/configure
index c02b197..ba0c99b 100755
--- a/configure
+++ b/configure
@@ -474,6 +474,12 @@ mobical:
 	\$(MAKE) -C "\$(CADICALBUILD)" mobical
 update:
 	\$(MAKE) -C "\$(CADICALBUILD)" update
+shared:
+	\$(MAKE) -C "\$(CADICALBUILD)" libcadical.so
+dynamiclib:
+	\$(MAKE) -C "\$(CADICALBUILD)" libcadical.dylib
+dll:
+	\$(MAKE) -C "\$(CADICALBUILD)" cadical.dll
 .PHONY: all cadical clean mobical test update
 EOF
 
diff --git a/makefile.in b/makefile.in
index 8fd957f..ad422a1 100644
--- a/makefile.in
+++ b/makefile.in
@@ -49,6 +49,15 @@ mobical: mobical.o libcadical.a makefile $(LIBS)
 libcadical.a: $(OBJ) makefile
 	ar rc $@ $(OBJ)
 
+libcadical.so: $(OBJ) makefile
+	$(COMPILE) -shared -o $@ $(OBJ)
+
+libcadical.dylib: $(OBJ) makefile
+	$(COMPILE) -shared -o $@ $(OBJ)
+
+cadical.dll: $(OBJ) makefile
+	$(COMPILE) -shared -o $@ $(OBJ) -lpsapi
+
 #--------------------------------------------------------------------------#
 
 # Note that 'build.hpp' is generated and resides in the build directory.
