diff --git a/utils/System.h b/utils/System.h
index 4e48fee..56d0afc 100644
--- a/utils/System.h
+++ b/utils/System.h
@@ -42,6 +42,7 @@ extern double memUsedPeak();        // Peak-memory in mega bytes (returns 0 for
 // Implementation of inline functions:
 
 #if defined(_MSC_VER) || defined(__MINGW32__)
+#include <sys/time.h>
 #include <time.h>
 
 static inline double Glucose::cpuTime(void) { return (double)clock() / CLOCKS_PER_SEC; }
